snakes_cafe.py

```diff
@@ -0,0 +1,130 @@
+from textwrap import dedent
+import sys
+
+
+MENU = [
+  {'Appetizers': {'1': 'Wings','2': 'Cookies','3':'Spring Rolls'}},
+  {'Entrees': {'4':'Salmon','5':'Steak','6':'Meat Tornado','7':'A Literal Garden'}},
+  {'Desserts': {'8':'Ice Cream','9':'Cake','10':'Pie'}},
+  {'Drinks': {'11':'Coffee','12':'Tea','13':'Blood of the Innocent'}},
+]
+AVAILABLE_ITEMS = {}
+
+for category in MENU:
+    for key,value in category.items():
+      for key2,value2 in value.items():
+        AVAILABLE_ITEMS[key2]=value2
+
+ORDER = {}
+
+
+def print_menu(menu):
+  output='\n'
+  for category in menu:
+    for key,value in category.items():
+      output += key + '\n'
+      output += f'''{'-' * len(key)}\n'''
+      for key2,value2 in value.items():
+          output += key2 + '. ' + value2 + '\n'
+    output += '\n'
+  return output
+
+
+def greeting():
+  """Function which will greet user when the application executes for the first time.
+  """
+  ln_one = 'Welcome to the Snakes Cafe! '
+  ln_two = 'Please see our menu below.'
+  ln_three = '**'
+  ln_four = 'To quit at any time, type "quit"'
+  ln_four_2 = 'To see your full order, type "show"'
+  ln_five = '**   What would you like to order?    **'
+  ln_six = '** (Please enter number next to item) **'
+
+  width = max(len(ln_one),len(ln_two),len(ln_three),len(ln_four),len(ln_four_2),len(ln_five)) + 8
+
+  print(dedent(f'''\n
+    {'*' * width}
+    {'**' + (' ' * ((width - 4 - len(ln_one)) // 2)) + ln_one + (' ' * ((width - 4 - len(ln_one)) // 2)) + '**'}
+    {'**' + (' ' * ((width - 4 - len(ln_two)) // 2)) + ln_two + (' ' * ((width - 4 - len(ln_two)) // 2)) + '**'}
+    {'**' + (' ' * ((width - 4 - len(ln_three)) // 2)) + ln_three + (' ' * ((width - 4 - len(ln_three)) // 2)) + '**'}
+    {'**' + (' ' * ((width - 4 - len(ln_four)) // 2)) + ln_four + (' ' * ((width - 4 - len(ln_four)) // 2)) + '**'}
+    {'**' + (' ' * ((width - 4 - len(ln_four_2)) // 2)) + ln_four_2 + (' ' * ((width - 4 - len(ln_four_2)) // 2)) + '**'}
+    {'*' * width}
+    \n
+    {print_menu(MENU)}
+    {'*' * len(ln_six)}
+    {ln_five}
+    {ln_six}
+    {'*' * len(ln_six)}
+  '''))
+
+def check_input(user_in):
+  if user_in.lower() == 'quit':
+    exit()
+    return
+  
+  if user_in == 'show':
+    return 'show'
+
+  elif(user_in in AVAILABLE_ITEMS):
```

Comment:
I would change this behavior. Entry by item number is much less user-friendly, and kind of circumvents issues you were intended to handle (like case sensitivity). If you don't refactor, you also might have a lot more trouble trying to work this behavior into labs later this week.

Reply:
While the code is far from being as elegant as it should be, I'd disagree with "much less user-friendly". Is it easier to type 'Blood of the Innocent' or '11'? Also, implementing case-sensitivity is pretty trivial; I thought this approach would be a little more challenging for me. Additionally, this gets around potential misspellings from the user.

I'll implement entry by name as an option (handling case-sensitivity) as well for today's lab.